extension/main.js

```diff
@@ -16,13 +16,31 @@
 	}
 
 	function update() {
-		gitHubNotifCount(function (count) {
+		gitHubNotifCount(function (count, interval) {
+			var period = 1;
+			if (interval !== GitHubNotify.settings.get('interval')) {
+				GitHubNotify.settings.set('interval', interval);
+				period = Math.ceil(interval / 60);
+				if (period < 1) {
+					period = 1;
+				}
+				chrome.alarms.clearAll(function () {
+					chrome.alarms.create({periodInMinutes: period});
+					chrome.alarms.onAlarm.addListener(update);
+					chrome.runtime.onMessage.addListener(update);
+				});
+			}
 			if (count < 0) {
```

Comment:
Scratch that. In fact, all you need to do is get rid of the `String()` that wraps `data.length` in `api.js`.